### PR TITLE
Replace duplicate #unauthorized method with module inclusion

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

I found it while working on https://github.com/openfoodfoundation/openfoodnetwork/pull/7433. 

No reason to risk diverging their implementations. The one in the `Auth` module and this one are exact copies.

#### What should we test?

Green build.

#### Release notes
Replace duplicate #unauthorized method with module inclusion
Changelog Category: Technical changes

### CODE REVIEW

File: app/controllers/spree/admin/base_controller.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ class BaseController < ApplicationController
       layout '/spree/layouts/admin'
 
       include I18nHelper
+      include Core::ControllerHelpers::Auth
```
Comment (Matt-Yorkley) [inline app/controllers/spree/admin/base_controller.rb:10]: This is already included in the parent `ApplicationController`, so we shouldn't need this line.

File: app/controllers/spree/admin/base_controller.rb [hunk 2]
```diff
@@ -21,18 +22,6 @@ def warn_invalid_order_cycles
         flash[:notice] = warning if warning.present?
       end
 
-      # This is in Spree::Core::ControllerHelpers::Auth
-      # But you can't easily reopen modules in Ruby
-      def unauthorized
-        if spree_current_user
-          flash[:error] = t(:authorization_failure)
```
Comment (Matt-Yorkley) [inline app/controllers/spree/admin/base_controller.rb:28]: There's a failing spec related to this line. The error uses `t(:authorization_failure)` here but in the module version it uses `Spree.t(:authorization_failure)` which is sightly different. We could maybe update the module version to use the non-spree translation and delete the I18n key?

Comment (sauloperez) [reply app/controllers/spree/admin/base_controller.rb:28]: done!

I'm thinking that removing the whole Spree.t and its keys wouldn't be too difficult but that would require syncing all the matching locales as well.

**Review by Matt-Yorkley — APPROVED** [review]
:muscle:

**Review by andrewpbrett — APPROVED** [review]

**Pull request merged**